src/main/java/redis/clients/jedis/JedisShardInfo.java

```diff
@@ -91,9 +102,21 @@ public void setTimeout(int timeout) {
   public String getName() {
     return name;
   }
+  
+  public int getDb() {
+    return db;
+  }
 
   @Override
   public Jedis createResource() {
-    return new Jedis(this);
+    Jedis jedis = new Jedis(this);
```

Comment:
@HeartSaVioR actually we will need to create a `setDb` in `BinaryClient`. This way we can configure the db and let the `connect` method use it.
I'm thinking about changing `BinaryJedis` constructor to receive the `db` and then call `setDb`. 
It will look like this:
```
public class BinaryJedis {
 public BinaryJedis(final JedisShardInfo shardInfo, int db) {
    client = new Client(shardInfo.getHost(), shardInfo.getPort());
    client.setTimeout(shardInfo.getTimeout());
    client.setPassword(shardInfo.getPassword());
    client.setDb(db);
  }
}
```

What do you think about it @HeartSaVioR?